llpc/tool/amdllpc.cpp

```diff
@@ -550,22 +546,21 @@ int main(int argc, char *argv[]) {
 
   auto inputSpecsOrErr = parseAndCollectInputFileSpecs(expandedInputFiles);
   if (Error err = inputSpecsOrErr.takeError()) {
-    reportError(std::move(err));
-    result = Result::ErrorInvalidValue;
+    result = reportError(std::move(err));
     return EXIT_FAILURE;
```

Comment:
I see `EXIT_FAILURE` and `EXIT_SUCCESS` being returned, but is it possible to explicitly return a typed result instead, e.g. like ErrorSuccess? Is there an automatic type conversion going on currently?

Reply:
`EXIT_SUCCESS` and `_FAILURE` are defined in the standard library: https://en.cppreference.com/w/cpp/utility/program/EXIT_status.
I think this comes from POSIX and is a standard practice to return them from main.

Reply:
Ah, I got confused by GitHub and I thought this was a different function, not `main`. Totally fine then.